src/mongoose_rabbit_worker.erl

```diff
@@ -0,0 +1,298 @@
+%%==============================================================================
+%% Copyright 2018 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc
+%% Generic RabbitMQ worker. The module is responsible for interacting with
+%% RabbitMQ server. Worker holds its own AMQP connection and channel in its
+%% state.
+%% @end
+%%==============================================================================
+
+-module(mongoose_rabbit_worker).
+-author('kacper.mentel@erlang-solutions.com').
+
+-include_lib("mongooseim/include/mongoose.hrl").
+
+-behaviour(gen_server).
+
+-export([start_link/0, list_metrics/0]).
+
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2]).
+
+-type state() :: #{amqp_client_opts := mongoose_amqp:network_params(),
+                   connection := pid(),
+                   channel := pid(),
+                   host := binary(),
+                   confirms := boolean(),
+                   max_queue_len := non_neg_integer() | infinity}.
+
+-type worker_opts() :: state().
+
+-define(REPLY_REQ_DROPPED(State), {reply, request_dropped, State}).
```

Comment:
These macros are used in only 3 places and given their length, there is no gain from introducing them.